libs/ui/test/expect_text_with_icon.ts

```diff
@@ -0,0 +1,16 @@
+import { screen, within } from './react_testing_library';
+
+export async function expectTextWithIcon(
+  text: string,
+  icon: string
+): Promise<void> {
+  const textElement = await screen.findByText(text);
+  expect(
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    within(textElement.closest('p')!)
```

Comment:
nit: this `closest('p')` might be a too-specific/brittle assumption for a general test helper - maybe `within(textElement.parentNode)`? (although, i guess the text could be nested in `span`s or broken up across elements)

Reply:
Yeah, I agree. I didn't think of it as a general purpose test helper, just a helper for the diagnostics page format. But I was repeating it and wanted to extract it, which put it in the `test` folder. Do you think a scoping comment would make it more acceptable?

Reply:
gotcha -- maybe move this file into the `src/diagnostics/` package? seems like it's only used there at the moment

Reply:
If I can, I'd like to avoid the pattern of having test helper files in `src`, which either need to be excluded from coverage or be `.test.tsx` files with exports. I've seen (and used) both but using the `test` directory seems preferable. I just added a comment, which will hopefully act as a disclaimer.